packages/block-editor/src/components/block-inspector/edit-contents.js

```diff
@@ -18,23 +21,66 @@ export default function EditContents( { clientId } ) {
 		stopEditingContentOnlySection,
 	} = useContentOnlySectionEdit( clientId );
 
+	const { block, onNavigateToEntityRecord } = useSelect(
+		( select ) => {
+			const { getBlock, getSettings } = select( blockEditorStore );
+			return {
+				block: getBlock( clientId ),
+				onNavigateToEntityRecord:
+					getSettings().onNavigateToEntityRecord,
+			};
+		},
+		[ clientId ]
+	);
+
 	if ( ! isWithinSection && ! isWithinEditedSection ) {
 		return null;
 	}
 
+	const blockAttributes = block?.attributes || {};
+
+	// Synced patterns and template parts should navigate to the isolated editor
+	const isSyncedPattern = isReusableBlock( block );
+	const isTemplatePartBlock = isTemplatePart( block );
+	const shouldNavigateToIsolatedEditor =
+		( isSyncedPattern || isTemplatePartBlock ) && onNavigateToEntityRecord;
+
+	const handleClick = () => {
+		if ( ! editedContentOnlySection ) {
+			if ( shouldNavigateToIsolatedEditor ) {
```

Comment:
With the existing code, I think it could also be simplified to this, and the `shouldNavigateToIsolatedEditor` variable deleted:
```suggestion
			if ( onNavigateToEntityRecord ) {
```

If you decide to make other changes to the code then `shouldNavigateToIsolatedEditor` might still be needed.